api/v1/cluster_types.go

```diff
@@ -1066,6 +1068,10 @@ type AzureCredentials struct {
 	// A shared-access-signature to be used in conjunction with
 	// the storage account name
 	StorageSasToken *SecretKeySelector `json:"storageSasToken,omitempty"`
+
+	// Use the Azure AD based authentication without providing explicitly the keys.
+	// +optional
+	InheritFromAzureAD bool `json:"inheritFromAzureAD"`
```

Comment:
I agree, can be also `EnableAzureADWorkLoadIdentity`. I've initially used used `InheritFromAzureAD` which is similar to `InheritFromIAMRole` that we have for AWS, but I agree that we can probably use a more meaningful name that refers to the related Azure feature.

Reply:
agreed when the config field maps to the underlying cloud capability then its lot clearer to the user without us needing additional documentation, otherwise we need to explicitly state that `InHeritFromAzureAD` uses `workloadIdentity` under the hood.